### PR TITLE
Handling no profile pic for the user

### LINKED ISSUE #261: fetching an undefined resource
The console shows a 404 error for logged-in users without a profile picture.

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
### Issue
- https://github.com/Real-Dev-Squad/website-www/issues/261

### What is the change?
- Added if condition in user.js for the profile picture src.

### Is Development Tested?

- [x] Yes
- [ ] No

### Before Change
image

### After Change
image

### CODE REVIEW

**Review by prakashchoudhary07 — DISMISSED** [review]
Nice :tada: :tada:

**Review by rohan09-raj — DISMISSED** [review]
LGTM 💯

File: js/analytics.js [hunk 1]
```diff
@@ -1 +1,43 @@
-!function (t, e) { var o, n, p, r; e.__SV || (window.posthog = e, e._i = [], e.init = function (i, s, a) { function g(t, e) { var o = e.split("."); 2 == o.length && (t = t[o[0]], e = o[1]), t[e] = function () { t.push([e].concat(Array.prototype.slice.call(arguments, 0))) } } (p = t.createElement("script")).type = "text/javascript", p.async = !0, p.src = s.api_host + "/static/array.js", (r = t.getElementsByTagName("script")[0]).parentNode.insertBefore(p, r); var u = e; for (void 0 !== a ? u = e[a] = [] : a = "posthog", u.people = u.people || [], u.toString = function (t) { var e = "posthog"; return "posthog" !== a && (e += "." + a), t || (e += " (stub)"), e }, u.people.toString = function () { return u.toString(1) + ".people (stub)" }, o = "capture identify alias people.set people.set_once set_config register register_once unregister opt_out_capturing has_opted_out_capturing opt_in_capturing reset isFeatureEnabled onFeatureFlags".split(" "), n = 0; n < o.length; n++)g(u, o[n]); e._i.push([i, s, a]) }, e.__SV = 1) }(document, window.posthog || []); posthog.init('phc_SRQcmzRDT5aNlAEkcrGgU2FqxKRaec66c7tdrntqAbs', { api_host: 'https://app.posthog.com' })
```
Comment (ankushdharkar) [inline js/analytics.js:1]: Please revert these changes. It's library code.

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: js/user.js [hunk 2]
```diff
@@ -19,7 +19,11 @@ const setUserGreeting = (username, firstName, userProfilePicture) => {
     const userImgURL = userProfilePicture;
 
     userImgEl.forEach((element) => {
-      element.src = userImgURL;
+      if (userImgURL) {
+        element.src = userImgURL;
+      } else {
+        element.src = '/img/profile.png';
+      }
```
Comment (ankushdharkar) [inline js/user.js:26]:
```suggestion
      element.src = userImgURL || '/img/profile.png';
```

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: js/user.js (hunk 2 above)
Comment (RitikJaiswal75) [reply js/user.js]: Done

File: js/user.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ const setUserGreeting = (username, firstName, userProfilePicture) => {
     const userImgURL = userProfilePicture;
 
     userImgEl.forEach((element) => {
-      element.src = userImgURL;
+      element.src = userImgURL || '/img/profile.png';
```
Comment (harshith-venkatesh) [inline js/user.js:22]: Is the purpose of this statement to ensure if the userImgURL is undefined or null then use /img/profile.png?

Do you think this would be helpful
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

**Review by harshith-venkatesh — CHANGES_REQUESTED** [review]

File: js/user.js (hunk 3 above)
Comment (RitikJaiswal75) [reply js/user.js:22]: I did think of that but Ankush mentioned the specified code

Comment (harshith-venkatesh) [reply js/user.js:22]: Ok, do you know any specific reason for going ahead with this?

Comment (harshith-venkatesh) [reply js/user.js:22]: Good discussion, We came to conclusion to go ahead with || operator here as we might as well receive `''`

**Review by harshith-venkatesh — APPROVED** [review]

**Review by rohan09-raj — APPROVED** [review]
Approving the changes
